rmf_fleet_adapter_python/scripts/rmf_fleet_adapter_python/test_loop.py

```diff
@@ -0,0 +1,190 @@
+                                        profile=profile)
+
+    # Manages delivery or loop requests
+    adapter = adpt.MockAdapter("TestDeliveryAdapter")
+    fleet = adapter.add_fleet(fleet_name, robot_traits, test_graph)
+    fleet.accept_delivery_requests(lambda x: True)
+
+    cmd_node = Node("RobotCommandHandle")
+
+    # Test compute_plan_starts, which tries to place the robot on the navgraph
+    # Your robot MUST be near a waypoint or lane for this to work though!
+    starts = plan.compute_plan_starts(test_graph,
+                                      map_name,
+                                      [[-10.0], [0.0], [0.0]],
+                                      adapter.now())
+    assert [x.waypoint for x in starts] == [3], [x.waypoint for x in starts]
+
+    # Alternatively, if you DO know where your robot is, place it directly!
+    starts = [plan.Start(adapter.now(),
+                         0,
+                         0.0)]
+
+    # Lambda to insert an adapter
+    def updater_inserter(handle_obj, updater):
+        handle_obj.updater = updater
+
+    # Manages and executes robot commands
+    robot_cmd = MockRobotCommand(cmd_node, test_graph)
+
+    fleet.add_robot(robot_cmd,
+                    "T0",
+                    profile,
+                    starts,
+                    partial(updater_inserter, robot_cmd))
+
+    # INIT TASK SUMMARY OBSERVER ==============================================
+    observer = TaskSummaryObserver()
+
+    # FINAL PREP ==============================================================
+    rclpy_executor = SingleThreadedExecutor()
+    rclpy_executor.add_node(cmd_node)
+    rclpy_executor.add_node(observer)
+
+    # GO! =====================================================================
+    adapter.start()
+
+    print("\n")
+    print("# SENDING SINGLE LOOP REQUEST ####################################")
+    print(test_graph_vis)
+    request = adpt.type.CPPLoopMsg(test_name,
+                                   fleet_name,
+                                   loop_count,
+                                   start_name,
+                                   finish_name)
+    observer.reset()
+    observer.add_task(test_name)
+    adapter.request_loop(request)
+
+    for i in range(1000):
+        if observer.all_tasks_complete():
+            print("Tasks Complete.")
+            break
+        rclpy_executor.spin_once(1)
+        # time.sleep(0.2)
+
+    results = observer.count_successful_tasks()
+    print("\n== DEBUG TASK REPORT ==")
+    print("Visited waypoints:", robot_cmd.visited_waypoints)
+    print(f"Sucessful Tasks: {results[0]} / {results[1]}")
+
+    assert results[0] == results[1], "Not all tasks were completed."
+
+    error_msg = "Robot did not take the expected route"
+    assert robot_cmd.visited_waypoints == [
+        0, 0, 5, 5, 6, 6, 7,
+        6, 5, 5, 8, 8, 10,
+        8, 5, 5, 6, 6, 7,
+        6, 5, 5, 8, 8, 10], error_msg
+
```

Comment:
Might also need to add these below, just to be safe.
```
cmd_node.destroy_node()
observer.destroy_node()
rclpy_executor.shutdown()
rclpy.shutdown()
```

Reply:
thank you for catching this!

Reply:
added this